src/Collectors/PublicPropertyFetchCollector.php

```diff
@@ -53,19 +53,23 @@ public function processNode(Node $node, Scope $scope): ?array
             return null;
         }
 
-        $propertyFetcherType = $scope->getType($node->var);
-        if (! $propertyFetcherType instanceof TypeWithClassName) {
```

Comment:
this kind of bugs are the reason why these instanceof checks are deprecated:

https://phpstan.org/blog/why-is-instanceof-type-wrong-and-getting-deprecated

Reply:
IDE support is the reason 🙏